### PR TITLE
[WebNN EP] Fix bug of padding in Op ConvTranspose

### PR DESCRIPTION
**Pull request by zesongw**
Get the dimensions of H and W according to the layout.

### CODE REVIEW

**Comment by zesongw** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc [hunk 1]
```diff
@@ -251,8 +251,14 @@ Status ConvOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const N
...
         ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get shape");
         for (size_t i = 0; i < 2; i++) {
-          total_padding[i] = strides[i] * (narrow<size_t>(input_shape[i + 1]) - 1) +
-                             output_padding[i] + ((kernel_shape[i] - 1) * dilations[i] + 1) - output_shape[i];
+          // Get the dimensions of H and W.
+          if (model_builder.GetPreferredLayout() == DataLayout::NHWC) {
+            total_padding[i] = strides[i] * (narrow<size_t>(input_shape[i + 1]) - 1) +
+                               output_padding[i] + ((kernel_shape[i] - 1) * dilations[i] + 1) - output_shape[i];
+          } else {
+            total_padding[i] = strides[i] * (narrow<size_t>(input_shape[i + 2]) - 1) +
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:259]: Shouldn't we at least `assert` in the `else` the layout here?

`assert(model_builder.GetPreferredLayout() == DataLayout::NCHW)`

Though, when would ONNX ever receive a model with NCHW in it? ONNX Conv only ever supports NCHW: https://onnx.ai/onnx/operators/onnx__Conv.html

Comment (zesongw) [reply onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:259]: ONNXRuntime can add transpose for EPs with preferred layout of NHWC.
https://github.com/microsoft/onnxruntime/blob/fc8631e2f11d85c84ab9cc711aacb9c589b6f71a/onnxruntime/core/framework/graph_partitioner.cc#L182
WebNN sets preferred_layout_ = DataLayout::NHWC for CPU backend.
https://github.com/microsoft/onnxruntime/blob/fc8631e2f11d85c84ab9cc711aacb9c589b6f71a/onnxruntime/core/providers/webnn/webnn_execution_provider.cc#L30

Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:259]: > ONNXRuntime can add transpose for EPs with preferred layout of NHWC.
👍

assert or comment in else branch appreciated.

**Review by fdwr — APPROVED** [review]

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc (hunk 1 above)
Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:261]: Oops, I clicked approve prematurely. Would still like to see this comment/assert added here.

**Review by fdwr — APPROVED** [review]
👍 Thanks Z.

**Pull request merged**